compass/landice/tests/slm/circ_icesheet/setup_mesh.py

```diff
@@ -0,0 +1,409 @@
+    rhoi = 910.0  # ice density
+
+    # Open the file, get needed dimensions
+    shutil.copy(mali_mesh_file, filename)
+    smbfile = NetCDFFile(filename, 'r+')
+
+    # Get variables
+    xCell = smbfile.variables['xCell']
+    yCell = smbfile.variables['yCell']
+    nCells = smbfile.dimensions['nCells'].size
+
+    # time interval for the forcing
+    t_array = np.arange(start_year, end_year, dt_year)
+    nt = len(t_array)
+
+    # initialize SMB value everywhere in the mesh
+    sfcMassBal = np.zeros((nt, nCells))
+
+    # Find center of domain
+    x0 = xCell[:].min() + 0.5 * (xCell[:].max() - xCell[:].min())
+    y0 = yCell[:].min() + 0.5 * (yCell[:].max() - yCell[:].min())
+    # calculate the radius of circ ice sheet
+    r = ((xCell[:] - x0) ** 2 + (yCell[:] - y0) ** 2) ** 0.5
+
+    if direction == 'vertical':
+        logger.info('creating a file that prescribes vertical smb')
+        smb = dhdt * rhoi / (365.0 * 24 * 60 * 60)
+        logger.info(f'prescribed smb is {smb} kg/m2/s')
+        indx = np.where(r <= r0)[0]
+        # for loop through time
+        for t in range(len(t_array)):
+            Ht = h0 + (dhdt * (t + 1))  # calculate a new height each time
+            logger.info(f'At time {start_year + dt_year}, \
+                        new height will be {Ht} km')
+            sfcMassBal[t, indx] = smb  # assign sfcMassBal
+        smbfile.variables['sfcMassBal'][:, :] = sfcMassBal[:, :]
+
+    elif direction == 'horizontal':
+        logger.info('creating a file that prescribes horizontal smb')
+        smb = -1 * ((10000.0) * rhoi / (365.0 * 24.0 * 60.0 * 60.0))
+        print(f'prescribed smb is {smb} kg/m2/s')
+        # for loop through time
+        for t in range(len(t_array)):
+            # calculate a new radius at each time
+            Rt = r0 + (drdt * (t + 1))
+            if (Rt < 0):
+                Rt = 0.0
+                sfcMassBal[t, :] = 0.0
+                print(f'At time {start_year+dt_year*t}, \
+                      new radius will be {Rt} km')
+            else:
+                indx = np.where(r >= Rt)[0]
+                sfcMassBal[t, indx] = smb
+                print(f'At time {start_year+dt_year*t}, \
+                      new radius will be {Rt/1000} km')
+        smbfile.variables['sfcMassBal'][:, :] = sfcMassBal[:, :]
+
+    # add xtime variable
+    smbfile.createDimension('StrLen', 64)
+    xtime = smbfile.createVariable('xtime', 'S1', ('Time', 'StrLen'))
+
+    # initialize SMB value everywhere in the mesh
+    xtime_str = []
+    for t_index in range(len(t_array)):
+        yr = start_year + (t_index * dt_year)
+        xtime_str = f'{int(yr)}-01-01_00:00:00'.ljust(64)
+        xtime_char = netCDF4.stringtochar(np.array([xtime_str], 'S64'),
+                                          encoding='utf-8')
+        xtime[t_index, :] = xtime_char
+
+    smbfile.close()
+
+
+def _build_mapping_files(config, logger, res, nglv, mali_mesh_file):
```

Comment:
Is this identical (or very similar) to the version in the ISMIP6-2300 SLM test case PR (#749)?  If so, we should move this to a compass landice framework module and call it from both places.  That does not need to need to be in the scope of your revisions - I can do that in a follow up PR.

Reply:
Sounds good - I'll leave this comment unresolved so you can follow up with it.